packages/backend-modules/access/lib/grants.js

```diff
@@ -432,13 +449,19 @@ const invalidate = async (grant, reason, t, pgdb, mail) => {
         })
       }
 
+      const perks = revokePerks(grant, recipient, campaign, pgdb)
```

Comment:
```suggestion
      const perks = await revokePerks(grant, recipient, campaign, pgdb)
```

We do need to resolve this Promise, too, before we can proceed.